Dockerfile.multistage

```diff
@@ -0,0 +1,28 @@
+# An example using multi-stage image builds
+
+# First, build the application in the `/app` directory
+FROM ghcr.io/astral-sh/uv:python3.12-bookworm-slim AS builder
+WORKDIR /app
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-install-project
```

Comment:
We need to define the link mode:

```
#11 1.303 warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
#11 1.303          If the cache and target directories are on different filesystems, hardlinking may not be supported.
#11 1.303          If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
```

Reply:
Will do separate from this change.